addons/hr_expense/models/hr_expense.py

```diff
@@ -0,0 +1,317 @@
+            self.product_uom_id = self.product_id.uom_id
+
+    @api.onchange('product_uom_id')
+    def _onchange_product_uom_id(self):
+        if self.product_id:
+            if self.product_uom_id.category_id != self.product_id.uom_id.category_id:
+                raise UserError(_('Selected Unit of Measure does not belong to the same category as the product Unit of Measure'))
+
+    @api.onchange('employee_id')
+    def _onchange_employee_id(self):
+        self.department_id = self.employee_id.department_id
+
+    def _add_follower(self, employee_id):
+        user_ids = []
+        employee = self.env['hr.employee'].browse(employee_id)
+        if employee.user_id:
+            user_ids.append(employee.user_id.id)
+        if employee.parent_id:
+            user_ids.append(employee.parent_id.user_id.id)
+        if employee.parent_id != employee.department_id.manager_id:
+            user_ids.append(employee.department_id.manager_id.user_id.id)
+        self.message_subscribe_users(user_ids = user_ids)
+
+    @api.model
+    def create(self, vals):
+        employee_id = vals.get('employee_id')
+        hr_expense = super(HrExpense, self).create(vals)
+        if employee_id:
+            hr_expense._add_follower(employee_id)
+        return hr_expense
+
+    @api.multi
+    def write(self, vals):
+        employee_id = vals.get('employee_id')
+        hr_expense = super(HrExpense, self).write(vals)
+        if employee_id:
+            self._add_follower(employee_id)
+        return hr_expense
+
+    @api.multi
+    def unlink(self):
+        for expense in self:
+            if expense.state not in ['draft', 'refuse']:
+                raise UserError(_('You can only delete draft or refused expenses!'))
+        return super(HrExpense, self).unlink()
+
+    @api.multi
+    def submit_expenses(self):
+        if any(expense.state != 'draft' for expense in self):
+            raise UserError(_("You can only submit draft expenses!"))
+        self.write({'state': 'submit'})
+
+    @api.multi
+    def approve_expenses(self):
+        self.write({'state': 'approve'})
+
+    @api.multi
+    def refuse_expenses(self, reason):
+        self.write({'reason': reason, 'state': 'cancel'})
+
+    @api.multi
+    def paid_expenses(self):
+        self.write({'state': 'done'})
+
+    @api.multi
+    def reset_expenses(self):
+        return self.write({'state': 'draft'})
+
+    @api.multi
+    def _track_subtype(self, init_values):
+        self.ensure_one()
+        if 'state' in init_values and self.state == 'approve':
+            return 'hr_expense.mt_expense_approved'
+        elif 'state' in init_values and self.state == 'submit':
+            return 'hr_expense.mt_expense_confirmed'
+        elif 'state' in init_values and self.state == 'cancel':
+            return 'hr_expense.mt_expense_refused'
+        return super(HrExpense, self)._track_subtype(init_values)
+
+    def _line_get_convert(self, x, part, date):
```

Comment:
what is x, part, date ? can have meaning full name like part => partner ?
can you explain in doc string what's purpose of this method ? give meaning full method name.